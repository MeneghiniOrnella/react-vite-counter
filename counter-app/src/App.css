.App {
  text-align: center;
}

span{
  color: rgb(25, 109, 74);
}

button {
  font-size: calc(10px + 2vmin);
}
