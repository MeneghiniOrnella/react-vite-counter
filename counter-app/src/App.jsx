import { useState } from 'react';
import PropTypes from 'prop-types';
//import './App.css';

export const CounterApp = ({ value }) => {
  const [counter, setCounter] = useState(value)
  
  return (
    <div className="App">
      <h1>Counter: <span>{ counter }</span></h1>
      <div>
        <button onClick={ () => setCounter(counter +1) }>+ 1</button>
        <button onClick={ () => setCounter(value) }>Reset</button>
        <button onClick={ () => setCounter(counter -1) }>- 1</button>
      </div>
    </div>
  )
}


CounterApp.propTypes = {
  value: PropTypes.number.isRequired
}
