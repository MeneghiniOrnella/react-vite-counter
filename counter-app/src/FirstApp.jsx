import PropTypes from "prop-types";

export const FirstApp = ({ title, subTitle, name }) => {
    return(
        <div>
            <h1 data-testid="test-title">{ title }</h1>
            <h2>{ subTitle }</h2>
            <p>{ name }</p>
        </div>
    )
}

FirstApp.propTypes = {
    title: PropTypes.string.isRequired,
    subTitle: PropTypes.string,
}

FirstApp.defaultProps = {
    name: 'name',
    subTitle: 'subtitle',
    //title: 'Hello world'
}