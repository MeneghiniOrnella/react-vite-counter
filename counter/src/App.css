.App {
  text-align: center;
}

span{
  color: rgb(25, 109, 74);
}