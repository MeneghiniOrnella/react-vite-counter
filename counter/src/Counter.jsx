import { useState } from 'react';
import './App.css';

function Counter({ value }) {
  const [counter, setCounter] = useState(value);
  return (
    <div className="App">
      <h1>Counter: <span className='counter'>{ counter }</span></h1>
      <div>
        <button onClick={ () => setCounter(counter +1) }>+ 1</button>
        <button onClick={ () => setCounter(value) }>Reset</button>
        <button onClick={ () => setCounter(counter -1) }>- 1</button>
      </div>
    </div>
  );
}

export default Counter;
