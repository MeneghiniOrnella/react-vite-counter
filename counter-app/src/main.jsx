import React from 'react';
import ReactDOM from 'react-dom/client';
import {CounterApp} from './App.jsx';
import {FirstApp} from './FirstApp';
import './index.css';

ReactDOM.createRoot(document.getElementById('root')).render(
  <React.StrictMode>
    <CounterApp value={ 5 } />
    <hr />
    <FirstApp title="Hello world" />
  </React.StrictMode>
)
