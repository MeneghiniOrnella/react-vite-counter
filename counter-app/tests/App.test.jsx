import { fireEvent, render, screen } from '@testing-library/react';
import { CounterApp } from '../src/App';

describe('pruebas en CounterApp', () => {
    const initialValue = 5;

     test('debe de hacer match con snapshot', () => {
        const { container } = render( <CounterApp value={ initialValue } /> );
        expect(container).toMatchSnapshot();
    }); 

    test('debe de mostrar initialValue = 5', () => {
        render( <CounterApp value={ 5 } /> );
        expect(screen.getByText(5)).toBeTruthy();
    });

    test('incrementar +1', () => {
        render( <CounterApp value={ initialValue } /> );
        fireEvent.click(screen.getByText('+ 1'));
        expect(screen.getByText('6')).toBeTruthy();
    });
    test('decrementar -1', () => {
        render( <CounterApp value={ initialValue } /> );
        fireEvent.click(screen.getByText('- 1'));
        expect(screen.getByText('4')).toBeTruthy();
    });
})