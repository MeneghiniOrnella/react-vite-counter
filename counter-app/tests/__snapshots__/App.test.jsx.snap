// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`pruebas en CounterApp debe de hacer match con snapshot 1`] = `
<div>
  <div
    class="App"
  >
    <h1>
      Counter: 
      <span>
        5
      </span>
    </h1>
    <div>
      <button>
        + 1
      </button>
      <button>
        Reset
      </button>
      <button>
        - 1
      </button>
    </div>
  </div>
</div>
`;
